global/scripts/MapVolumeToCIFTI.sh

```diff
@@ -0,0 +1,179 @@
+StudyFolder="${1}"
+Subject="${2}"
+fMRIName="${3}"
+CorticalLUT="${4}"
+SubCorticalLUT="${5}"
+Caret7_Command="${6}"
+LowResMesh="${7}"
+RegNames="${8}"
+SmoothingFWHM="${9}"
+FinalfMRIResolution="${10}"
+BrainOrdinatesResolution="${11}"
+Inputs="${12}" 
+Outputs="${13}"
+Names="${14}"
+Flag="${15}" #FALSE=Don't fix zeros, TRUE=Fix zeros
+DeleteIntermediates="${16}" #TRUE/FALSE
+VolExT="${17}"
+WorkingDir="${18}"
+
+Sigma=`echo "$SmoothingFWHM / ( 2 * ( sqrt ( 2 * l ( 2 ) ) ) )" | bc -l`
+
+RegNames=`echo ${RegNames} | sed 's/@/ /g'`
+
+Inputs=`echo ${Inputs} | sed 's/@/ /g'`
+Outputs=`echo ${Outputs} | sed 's/@/ /g'`
+Names=`echo ${Names} | sed 's/@/ /g'`
+
+
+AtlasFolder="${StudyFolder}/${Subject}/MNINonLinear"
+AtlasAtlasResultsFolder="${AtlasFolder}/Results/${fMRIName}"
+NativeFolder="${AtlasFolder}/Native"
+DownsampleFolder="${AtlasFolder}/fsaverage_LR${LowResMesh}k"
+ROIFolder="${AtlasFolder}/ROIs"
+
+if [ ${WorkingDir} = "NONE" ] ; then
+  WorkingDir="${AtlasAtlasResultsFolder}"
+fi
+
+  i=1
+  for File in ${Names} ; do
+    Input=`echo ${Inputs} | cut -d " " -f ${i}`
+    Output=`echo ${Outputs} | cut -d " " -f ${i}`
+    cp ${Input} ${WorkingDir}/${fMRIName}_${File}.${VolExT}
+
+    roiVolume="${AtlasAtlasResultsFolder}/RibbonVolumeToSurfaceMapping/goodvoxels.nii.gz"
+    if [ ! ${Flag} = "FALSE" ] ; then
+      ResampleFlag="-fix-zeros"
+      VolROI="-volume-roi $roiVolume"
+    else
+      ResampleFlag=""
+      VolROI=""
+    fi
+        
+    unset POSIXLY_CORRECT
+    if [ 1 -eq `echo "$BrainOrdinatesResolution == $FinalfMRIResolution" | bc -l` ] ; then
+	    #If using the same fMRI and grayordinates space resolution, use the simple algorithm to project bias field into subcortical CIFTI space like fMRI
+	    volumeIn="${WorkingDir}/${fMRIName}_${File}.${VolExT}"
+	    currentParcel="${ROIFolder}/ROIs.${BrainOrdinatesResolution}.nii.gz"
+	    newParcel="${ROIFolder}/Atlas_ROIs.${BrainOrdinatesResolution}.nii.gz"
+	    kernel="${Sigma}"
+	    volumeOut="${WorkingDir}/${File}_AtlasSubcortical.${VolExT}"
+	    $Caret7_Command -volume-parcel-resampling $volumeIn $currentParcel $newParcel $kernel $volumeOut ${ResampleFlag}
```

Comment:
This is the old way that we don't use anymore.  Why are we adding it back to the pipelines in global/scripts?

Reply:
This is the same script that Matt used to do this type of operation in `/media/myelin/brainmappers/Connectome_Project/YA_HCP_Final/Scripts/MergeAndMapDropouts.bat` so I just adapted it into the pipeline. How can this part be refined instead?

Reply:
The simple answer is to "hide" it in ICAFIX/scripts and live with it for feature generation only.  If we wanted to update it, we would use this method for subcortical data:

https://github.com/Washington-University/HCPpipelines/blob/master/fMRISurface/scripts/SubcorticalProcessing.sh

The current predilate amount on the -cifti-resample might be excessive, and take more time to run than the old way.